service/controller/release/resource/status/create.go

```diff
@@ -41,6 +43,41 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
+	// Doing this per-release isn't ideal, can we pass this list to each status reconcile somehow?
```

Comment:
I think the latest `operatorkit` provides some caching primitives. Is that what you were thinking?

Reply:
I looked around in `operatorkit` but can't tell from the docs and code I saw whether it's what I was thinking. The problem I see is that this code runs once for every release, so we wind up doing 5 API calls per release when chances are the clusters won't change between reconciling this release and the next one. It isn't a ton of waste, just not optimal. It would be nice if we could do this once per reconciling ALL the releases instead of per-resource.

Reply:
I don't know too much about it. Not a big deal anyways.